### PR TITLE
Switch wsconnector to internal socket client

### PR DESCRIPTION
**Pull request by AIIX**
- Introduces a custom socket asynchronous HomeAssistantClient that replaces HASS client implementation with minimal changes on connector side
- Fixes all bugs and errors encountered with using the HASS Client
- Removes the "use_websocket" config key as this is already discovered at runtime on instance setup and on startup for already configured instances
- Moves all debug behind a "enable_debug" configuration flag to keep the PHAL client logs clean

### CODE REVIEW

**Review by NeonDaniel — COMMENTED** [review]
Skimmed changes and the look good, couple bugs from testing (local wss connection):
- After turning off a device on or off in the UI, the new state isn't displayed until I go to a different view and come back (only noticed in grouped view, appeared to work in devices view)

Other less important notes:
- Camera previews work, but playback does not
- When I display grouped, the bottom bar is all `...`, could this be made scrollable?
- All of my rooms have `s` appended (i.e. "Hallways", "Kitchens", "Downstairss"

**Comment by AIIX** [comment]
This PR additionally now adds support for QR Code based oauth authentication for quicker setups. Additionally bugs mentioned above in the initial review have also now been fixed

**Review by NeonDaniel — CHANGES_REQUESTED** [review]
I have 19 areas displayed on the home screen, but only 8 are scrollable on the bottom bar before the right scroll arrow becomes grayed out; it stops scrolling after 4 presses if that's relevant

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ovos-plugin-manager>=0.0.1
-ovos-utils~=0.0.27a3
+ovos-utils>=0.0.27
```
Comment (NeonDaniel) [inline requirements.txt:2]: This could probably be `~=0.0.27` to prevent updating to an incompatible release (i.e. 1.0.0) in the future

**Review by NeonDaniel — APPROVED** [review]

**Pull request merged**